sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -0,0 +1,1225 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+
+from io import BytesIO
+from typing import (
+    Any, AnyStr, AsyncIterable, Dict, IO, Iterable, List, Optional, Tuple, Union,
+    TYPE_CHECKING
+)
+from urllib.parse import urlparse, quote, unquote
+
+from ._shared import encode_base64
+from ._shared.base_client import parse_query
+from ._shared.uploads import IterStreamer
+from ._shared.uploads_async import AsyncIterStreamer
+from ._shared.request_handlers import (
+    add_metadata_headers,
+    get_length,
+    read_length,
+    validate_and_format_range_headers)
+from ._shared.response_handlers import return_response_headers, return_headers_and_deserialized
+from ._generated.models import (
+    DeleteSnapshotsOptionType,
+    BlobHTTPHeaders,
+    BlockLookupList,
+    AppendPositionAccessConditions,
+    SequenceNumberAccessConditions,
+    QueryRequest,
+    CpkInfo)
+from ._serialize import (
+    get_modify_conditions,
+    get_source_conditions,
+    get_cpk_scope_info,
+    serialize_blob_tags_header,
+    serialize_blob_tags,
+    serialize_query_format, get_access_conditions
+)
+from ._deserialize import deserialize_blob_stream
+
+from ._encryption import modify_user_agent_for_encryption
+from ._models import (
+    BlobType,
+    BlobBlock,
+    BlobProperties,
+    QuickQueryDialect,
+    DelimitedJsonDialect,
+    DelimitedTextDialect,
+)
+from ._upload_helpers import _any_conditions
+
+if TYPE_CHECKING:
+    from urllib.parse import ParseResult
+
+_ERROR_UNSUPPORTED_METHOD_FOR_ENCRYPTION = (
+    'The require_encryption flag is set, but encryption is not supported'
+    ' for this method.')
```

Comment:
This is sort of a painful ask but can we move this into the encryption file? It doesn't really make sense here especially since `cotnainer_client` and encryption itself needs to reach into `blob_client_helpers` to get it which doesn't make a lot of sense.